app/cmd/rpc/mesh.go

```diff
@@ -383,7 +383,29 @@ func meshServicerNodeCheck(w http.ResponseWriter, r *http.Request, ps httprouter
 	}
 
 	chainsMap, err := app.PCA.QueryHostedChains()
+	if err != nil {
+		WriteErrorResponse(w, 400, err.Error())
+		return
+	}
+
 	health := app.PCA.QueryHealth(APIVersion)
+	latestHeight, err := app.PCA.QueryHeight()
+	if err != nil {
+		WriteErrorResponse(w, 400, err.Error())
+		return
+	}
+
+	paramReturn, err := app.PCA.QueryParam(latestHeight, "pos/BlocksPerSession")
```

Comment:
Ah, i see where you made the change now

Reply:
shouldn't you be getting the blocks per session for the "dispatch relay height", not the latest height?

Reply:
this is the "check" endpoint, so always return the "latest" height + block per session, I understand what u say but we are getting "latest" information on this endpoint.

Reply:
> dispatch relay height

for this we could been on a gap when we are talking about future sessions and we will not be asking for older sessions, because they will not be handled by the node, so always talking about "current" session (like the last one) + future sessions, that we cant be sure they are "real" so we just trust on what it is, and use last "known" blockPerSession for those "future" sessions

Reply:
I think we will need to revisit this (and maybe even Pocket core) if the network ever decides to adjust BlocksPerSession. It isn't something the network should do carelessly, and third party clients will need to adjust the behavior as well (i.e should blocks per session value be reflected before or after the end of session). Anyhow there isn't immediate need to anything here, as long as we both keep it in mind

Reply:
Yeah you are right, hopefully this will not be touched until v1 or not even there. I think for now is safe use a default and also ask the node for the value on the latest block, because that work for the "future" session and for the current one.